frontend/src/metabase/common/components/EntityPicker/components/ItemList/PersonalCollectionItemList.tsx

```diff
@@ -0,0 +1,64 @@
+import { useMemo } from "react";
+
+import { useCollectionListQuery } from "metabase/common/hooks";
+import type { Collection, SearchModelType } from "metabase-types/api";
+
+import type {
+  TypeWithModel,
+  TisFolder,
+  CollectionPickerItem,
+} from "../../types";
+
+import { ItemList } from "./ItemList";
+
+interface PersonalCollectionsItemListProps<TItem extends TypeWithModel> {
+  onClick: (val: TItem) => void;
+  selectedItem: TItem | null;
+  itemName: string;
+  isFolder: TisFolder<TItem>;
+  isCurrentLevel: boolean;
+}
+
+export const PersonalCollectionsItemList = <TItem extends TypeWithModel>({
+  onClick,
+  selectedItem,
+  itemName,
+  isFolder,
+  isCurrentLevel,
+}: PersonalCollectionsItemListProps<TItem>) => {
+  const { data: collections, isLoading } = useCollectionListQuery({
+    query: { "personal-only": true },
+  });
+
+  const topLevelPersonalCollections = useMemo(
+    () =>
+      getSortedTopLevelPersonalCollections(collections) as unknown as TItem[],
+    [collections],
+  );
+
+  return (
+    <ItemList
+      items={topLevelPersonalCollections}
+      isLoading={isLoading}
+      onClick={onClick}
+      selectedItem={selectedItem}
+      itemName={itemName}
+      isFolder={isFolder}
+      isCurrentLevel={isCurrentLevel}
+    />
+  );
+};
+
+const getSortedTopLevelPersonalCollections = (
+  personalCollections?: Collection[],
+): CollectionPickerItem[] | null =>
+  personalCollections
+    ?.filter(isRootPersonalCollection)
+    .map((collection: Collection) => ({
+      ...collection,
+      model: "collection" as SearchModelType,
+    }))
```

Comment:
I noodled a little and noticed this also works:

```suggestion
    .map((collection: Collection): CollectionPickerItem => ({
      ...collection,
      model: "collection",
    }))
```